src/CaNNOLeS.jl

```diff
@@ -623,15 +623,34 @@ function SolverCore.solve!(
   return stats
 end
 
-function newton_system(x, r, λ, Fx, rhs, LDLT, ρold, params, method, linsolve)
-  nvar = length(x)
-  nequ = length(r)
-  ncon = length(λ)
+"""
+    newton_system(nvar, nequ, ncon, rhs, LDLT, ρold, params, linsolve)
+
+Compute an LDLt factorization of the (`nvar + nequ + ncon`)-square matrix for the Newton system contained in `LDLT` with the method `linsolve`, i.e. `sparse(LDLT.rows, LDLT.cols, LDLT.vals, N, N)`.
+If the factorization fails, a new factorization is attempted with an increased value for the regularization ρ as long as it is smaller than `params[:ρmax]`.
+The factorization is then used to solve the linear system whose right-hand side is `rhs`.
+
+# Output
+
+- `d`: the solution of the linear system;
+- `ρ`: the last value of the regularization parameter;
+- `ρold`: the old regularization value. It is equal to `ρ` if `ρ <= params[:ρmax]`;
```

Comment:
This is not very clear. What is the "old" value?

Reply:
I tried to clarify this as: the value of the regularization parameter used in the previous successful factorization, or 0 if this is the first one;